plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/CompareWithRevisionAction.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.git.client.action;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.eclipse.che.ide.api.action.ActionEvent;
+import org.eclipse.che.ide.api.app.AppContext;
+import org.eclipse.che.ide.api.selection.Selection;
+import org.eclipse.che.ide.ext.git.client.GitLocalizationConstant;
+import org.eclipse.che.ide.ext.git.client.compare.revisionsList.RevisionListPresenter;
+import org.eclipse.che.ide.part.explorer.project.ProjectExplorerPresenter;
+import org.eclipse.che.ide.project.node.FileReferenceNode;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Action for comparing with revision.
+ *
+ * @author Igor Vinokur
+ */
+@Singleton
+public class CompareWithRevisionAction extends GitAction {
+    private final RevisionListPresenter    presenter;
+    private final ProjectExplorerPresenter projectExplorer;
+
+    @Inject
+    public CompareWithRevisionAction(RevisionListPresenter presenter,
+                                     AppContext appContext,
+                                     GitLocalizationConstant locale,
+                                     ProjectExplorerPresenter projectExplorer) {
```

Comment:
projectExplorer is needed also to give it to super
https://github.com/codenvy/che-plugins/blob/IDEX-3928/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/CompareWithRevisionAction.java#L41